presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -696,4 +731,285 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+            getDataPath(icebergTable.location()),
+            getFileFormat(icebergTable));
+    }
+
+    @Override
+    public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(ConnectorSession session,
+            ConnectorInsertTableHandle insertHandle,
+            Collection<Slice> fragments,
+            Collection<ComputedStatistics> computedStatistics,
+            List<ConnectorTableHandle> sourceTableHandles)
+    {
+        IcebergWritableTableHandle table = (IcebergWritableTableHandle) insertHandle;
+        org.apache.iceberg.Table icebergTable = transaction.table();
+        List<CommitTaskData> commitTasks = new ArrayList<>();
+
+        if (!fragments.isEmpty()) {
+            commitTasks = fragments.stream()
+                .map(slice -> commitTaskCodec.fromJson(slice.getBytes()))
+                .collect(toImmutableList());
+
+            Type[] partitionColumnTypes = icebergTable.spec().fields().stream()
+                .map(field -> field.transform().getResultType(
+                    icebergTable.schema().findType(field.sourceId())))
+                .toArray(Type[]::new);
+
+            AppendFiles appendFiles = transaction.newFastAppend();
+            for (CommitTaskData task : commitTasks) {
+                HdfsContext context = new HdfsContext(session, table.getSchemaName(), table.getTableName());
+                DataFiles.Builder builder = DataFiles.builder(icebergTable.spec())
+                        .withInputFile(new HdfsInputFile(new Path(task.getPath()), hdfsEnvironment, context))
+                        .withFormat(table.getFileFormat())
+                        .withMetrics(task.getMetrics().metrics());
+
+                if (!icebergTable.spec().fields().isEmpty()) {
+                    String partitionDataJson = task.getPartitionDataJson()
+                            .orElseThrow(() -> new VerifyException("No partition data for partitioned table"));
+                    builder.withPartition(PartitionData.fromJson(partitionDataJson, partitionColumnTypes));
+                }
+
+                appendFiles.appendFile(builder.build());
+            }
+
+            List<String> listOfStrings = sourceTableHandles.stream()
+                    .map(handle -> (IcebergTableHandle) handle)
+                    .filter(handle -> handle.getSnapshotId().isPresent())
+                    .map(handle -> handle.getSchemaTableName() + "=" + handle.getSnapshotId().get())
+                    .collect(Collectors.toUnmodifiableList());
+
+            // Update the 'dependsOnTables' property that tracks tables on which the materialized view depends and the corresponding snapshot ids of the tables
+            appendFiles.set(DEPENDS_ON_TABLES, Joiner.on(",").join(listOfStrings));
+            appendFiles.commit();
+        }
+        transaction.commitTransaction();
+        return Optional.of(new HiveWrittenPartitions(commitTasks.stream()
+            .map(CommitTaskData::getPath)
```

Comment:
I am using the checkstyle xml according to instructions from prestosql dev page. Despite running code-formatter, I get 4 space indent rather than 8. I manually fixed it. I will look into how I can fix it in intelliJ preferences.